File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        answer = [0 for _ in range(len(temperatures))]
        for i, temperature in enumerate(temperatures):
            while stack and stack[-1][1] < temperature:
                last = stack.pop()
                answer[last[0]] = i - last[0]
            
            stack.append((i, temperature))
        
        return answer

File: 0739-daily-temperatures/0739-daily-temperatures.rs
impl Solution {
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let n = temperatures.len();
        let mut stack: Vec<(i32, i32)> = vec![];
        let mut answer: Vec<i32> = vec![0; n];
        
        for (index, temperature) in temperatures.iter().enumerate() {
            while !stack.is_empty() && stack.last().unwrap().0 < *temperature {
                let (temp, j) = stack.pop().unwrap();
                answer[j as usize] = index as i32 - j;
            }
            stack.push((*temperature, index as i32));
        }
        return answer;
    }
}